### ISSUE TITLE
nutrient warning levels hard coded in JS 

### ISSUE THREAD

**Issue by duncanhealy** [codeforeurope/trinkwasser#5]
we have set 
```json
    'hardness': 180,
    'natrium': 120,
    'kalium': 12,
    'calcium': 600,
    'magnesium': 120,
    'chlorid': 240,
    'nitrat': 60,
    'sulfat': 1600
```
 in comarison.js, gauge.js 
but should be loading from EU regulations or other
http://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=CELEX:31998L0083&from=EN
? c.f.
https://www.epa.ie/pubs/advice/drinkingwater/2015_04_21_ParametersStandaloneDoc.pdf

Also other fields may be needed - eg microbateria etc

**Comment by duncanhealy** [comment]
can use the product and observation model to define the limits 

```javascript
  var product = require('../models/Product.js')
  var observation = require('../models/Observation.js')
    var uslimit = new product.model({
      name: 'US',
      observations: [
        new observation.model({value: 200, uom: mg_l, eqr: natrium}),
        new observation.model({value: null, uom: mg_l, eqr: kalium}),
        new observation.model({value: null, uom: mg_l, eqr: calcium}),
        new observation.model({value: 50, uom: mg_l, eqr: nitrat}),
        new observation.model({value: null, uom: mg_l, eqr: magnesium}),
        new observation.model({value: 0, uom: mg_l, eqr: fluorid}),
        new observation.model({value: 250, uom: mg_l, eqr: chlorid}),
        new observation.model({value: 250, uom: mg_l, eqr: sulfat}),
        new observation.model({value: null, uom: mg_l, eqr: hydrogene})
      ],
      sources: ['url_needed'],
      vendor: null
    })
```

**Comment by JohnRHarrington** [comment]
Who’s baseline values are those Duncan?

 

 


John Harrington


  <http://www.linkedin.com/img/signature/pic_plastic_cool_26x130.gif> 

Mobile: +44 748 222 1338

Mobile: +353 86 2518914

Email:  <mailto:john@innovatus.ie> john@innovatus.ie /  <mailto:john@jharrington.ie> john@jharrington.ie

www:  <http://www.innovatus.ie> www.innovatus.ie

IM: John_Harrington (Skype)
  <http://www.linkedin.com/img/signature/icon_in_blue_14x14.gif>   <http://ie.linkedin.com/in/jharrington1> http://ie.linkedin.com/in/jharrington1
 <https://twitter.com/intent/follow?original_referer=https%3A%2F%2Ftwitter.com%2Fabout%2Fresources%2Fbuttons&region=follow_link&screen_name=JHarrington100&tw_p=followbutton&variant=2.0> Follow me on Twitter

**Comment by duncanhealy** [comment]
so I've found this :) 
https://github.com/CivicVision/trinkwasser/issues/7

**Comment by JohnRHarrington** [comment]
Ok, but thats US based. IS there and I think you said it, a national or European recommended and minimum/maximum values statements somewhere? Could we see if someone from Trinity would like to help with this, as I know they have a public water testing facility there.

**Comment by duncanhealy** [comment]
Yes, EU standard referenced in Issue #5 above,  we can allow for water data to be compared to different standard and as there are likely to be new reference standard in future allow them to be plugged in easily

**Comment by milovanderlinden** [comment]
Would it be valid if we refer to individual values; like kalium=50 mg/l as "Observations"?

In my humble opinion; observations come in different flavours:
- From the label of a bottle
- From a measurement with a "te-be-created" device
- From (official) reports by water companies/offical entities.

The observations should have a timestamp and with regards to the bottled product to a sub-product. For instance; the Belgium Bottled Water company SPA will have 20cl, 33cl, 1L, 1.5L, 2L bottles, that may even be related to a certain batch(production) I would like to register as much as possible without it becoming an overkill, but with the possibility to start deiscovering trends through time. What are your ideas about that?

**Comment by duncanhealy** [comment]
Timestamp is there in the model it will be useful , I've observed differences in the mineral water data quoted from Wikipedia/ other source, I can imagine this being sent in via the API with some auth by relevant water plants in future

The actual limits will be defined by the relevant bodies and change over the years
